### ISSUE TITLE
[BUG] Volumes Stuck in Attach/Detach Loop when running on OpenShift/OKD

### ISSUE THREAD

**Issue by ArthurVardevanyan** [longhorn/longhorn#4988]
I see a few active Issues, relating too: "attaching/detaching loop with error", so this issue may be a duplicate, but I am not seeing the same errors in my logs. 

Might be Related too: https://github.com/longhorn/longhorn/issues/2241, but it seems to be that iSCSI is installed and functional.

I don't necessarily think this is a Longhorn issue.

## Describe the bug (🐛 if you encounter this issue)

(block device frontend)
- Regular Attach Mode: Volumes Stuck in Attach/Detach Loop 
  - (All Volumes, new and existing)
- Maintenance Attach Mode: No Issues

ISCSI Front End Seems to Attach Fine Maintenance/Normal

## To Reproduce
Upgraded from 4.11.0-0.okd-2022-11-19-050030 -> 4.11.0-0.okd-2022-12-03-100655

Or incept a cluster at or newer.
https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.11.0-0.okd-2022-12-02-145640

Install Longhorn, apply RBAC tweaks for OCP/OKD

1. Attach New or Existing Volume

## Log or Support bundle

Attached Support Bundle
[longhorn-support-bundle_00e45f14-1554-4313-bb2b-27dbf74ab4c7_2022-12-03T18-25-20Z.zip](https://github.com/longhorn/longhorn/files/10147101/longhorn-support-bundle_00e45f14-1554-4313-bb2b-27dbf74ab4c7_2022-12-03T18-25-20Z.zip)

Logs Should be relatively Clean, "grafana" was the volume I was testing after rebooting everything.

## Environment
 - Longhorn version: v1.3.2
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Kubectl
    -  Note: RBAC Modifications for OpenShift/OKD: https://github.com/ArthurVardevanyan/HomeLab/tree/production/kubernetes/longhorn/components/okd/rbac.yaml
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:  4.11.0-0.okd-2022-12-03-100655 (v1.24.6+5658434)
   -    https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.11.0-0.okd-2022-12-02-145640
 - Number of management node in the cluster: 3
 - Number of worker node in the cluster: 6
 - Node config
   - OS type and version: Fedora CoreOS 36  (Fedora CoreOS 411.36.4)
   - CPU per node: 4
   - Memory per node: ~13GB
   - Disk type(e.g. SSD/NVMe): NVME
   - Network bandwidth between the nodes: 1Gigabit Across "Zones", NVME Disk Speed in Same "Zone"
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): KVM
 - Number of Longhorn volumes in the cluster: 50 

## Additional context

Previous Working Version: 
https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.11.0-0.okd-2022-11-19-050030

I have also tried upgrading longhorn to the latest commit on the master branch: 
- https://github.com/longhorn/longhorn/commit/0e56cd1e9ae71182bd248ddeb535194815034d1b

This didn't help either

I was able to reproduce the issue on a new Sandbox Cluster.

It appears that:
- OKD 4.12.0-0.okd-2022-12-04-090858 
- Kubernetes 1.25.2
- Fedora CoreOS 37.20221116.10
- Longhorn 1.4.0-dev

works fine, so that may be an alternative solution when 1.4 is released.

**Comment by mantissahz** [comment]
Hi @ArthurVardevanyan, 
Thanks for reporting.
There is a waring log `FAIL to discover due to <nil>` in instance-manager-e-a931c2ec/engine-manager.log, and it might show it can not discover the target.
Could you login `instance-manager-e-a931c2ec` and show the information by command `iscsiadm -m node -T iqn.2019-10.io.longhorn:grafana-repo1 -p 10.130.4.67` with user `root`?

**Comment by ArthurVardevanyan** [comment]
I'm not finding any docs on how to log into the Instance Manager.

I did ```kubectl exec```, but not able to find the iscsiadm binary within the pod.
The Binary is on the host, should I SSH into the host and run the commands?

**Comment by innobead** [comment]
#1831

**Comment by mantissahz** [comment]
@ArthurVardevanyan sorry about the wrong info.
In pod `instance-manager-e-xxx`, use the command `tgtadm --lld iscsi --op show --mode target`
and on the volume attached node, use the command `iscsiadm -m discovery -t sendtargets -p 10.130.4.67` and `iscsiadm -m node -T iqn.2019-10.io.longhorn:grafana-repo1 -p 10.130.4.67`
yeah as mentioned at #1831, we do not support it completly yet.

**Comment by ArthurVardevanyan** [comment]
> @ArthurVardevanyan sorry about the wrong info. In pod `instance-manager-e-xxx`, use the command `tgtadm --lld iscsi --op show --mode target` and on the volume attached node, use the command `iscsiadm -m discovery -t sendtargets -p 10.130.4.67` and `iscsiadm -m node -T iqn.2019-10.io.longhorn:grafana-repo1 -p 10.130.4.67` yeah as mentioned at #1831, we do not support it completly yet.

No Worries.

Was doing so more testing and rebuilt my sandbox

Updated Bundle:
[longhorn-support-bundle_8684db96-ab85-457f-885a-3eabcccec963_2022-12-06T18-29-12Z.zip](https://github.com/longhorn/longhorn/files/10170366/longhorn-support-bundle_8684db96-ab85-457f-885a-3eabcccec963_2022-12-06T18-29-12Z.zip)

NOTE: After Running `sudo iscsiadm -m discovery -t sendtargets -p 10.129.2.10`, the pending volume attaches.


1. From `instance-manager-e-64aa96e4`

`tgtadm --lld iscsi --op show --mode target`

```log
Target 1: iqn.2019-10.io.longhorn:test
    System information:
        Driver: iscsi
        State: ready
    I_T nexus information:
    LUN information:
        LUN: 0
            Type: controller
            SCSI ID: IET     00010000
            SCSI SN: beaf10
            Size: 0 MB, Block size: 1
            Online: Yes
            Removable media: No
            Prevent removal: No
            Readonly: No
            SWP: No
            Thin-provisioning: No
            Backing store type: null
            Backing store path: None
            Backing store flags:
        LUN: 1
            Type: disk
            SCSI ID: IET     00010001
            SCSI SN: beaf11
            Size: 21475 MB, Block size: 512
            Online: Yes
            Removable media: No
            Prevent removal: No
            Readonly: No
            SWP: No
            Thin-provisioning: No
            Backing store type: longhorn
            Backing store path: /var/run/longhorn-test.sock
            Backing store flags:
    Account information:
    ACL information:
        ALL
```

2. `ssh -i ~/.ssh/id_ed25519 core@10.10.10.20` From the Node where Volume is being attached (Hosted on worker-2, attached to worker-0)

`sudo iscsiadm -m discovery -t sendtargets -p 10.129.2.10`

```log
10.128.2.10:3260,1 iqn.2019-10.io.longhorn:test
```

`sudo iscsiadm -m node -T iqn.2019-10.io.longhorn:test -p 10.128.2.10`

```log
# BEGIN RECORD 2.1.4

node.name = iqn.2019-10.io.longhorn:test
node.tpgt = 1
node.startup = automatic
node.leading_login = No
iface.iscsi_ifacename = default
iface.net_ifacename = <empty>
iface.ipaddress = <empty>
iface.prefix_len = 0
iface.hwaddress = <empty>
iface.transport_name = tcp
iface.initiatorname = <empty>
iface.state = <empty>
iface.vlan_id = 0
iface.vlan_priority = 0
iface.vlan_state = <empty>
iface.iface_num = 0
iface.mtu = 0
iface.port = 0
iface.bootproto = <empty>
iface.subnet_mask = <empty>
iface.gateway = <empty>
iface.dhcp_alt_client_id_state = <empty>
iface.dhcp_alt_client_id = <empty>
iface.dhcp_dns = <empty>
iface.dhcp_learn_iqn = <empty>
iface.dhcp_req_vendor_id_state = <empty>
iface.dhcp_vendor_id_state = <empty>
iface.dhcp_vendor_id = <empty>
iface.dhcp_slp_da = <empty>
iface.fragmentation = <empty>
iface.gratuitous_arp = <empty>
iface.incoming_forwarding = <empty>
iface.tos_state = <empty>
iface.tos = 0
iface.ttl = 0
iface.delayed_ack = <empty>
iface.tcp_nagle = <empty>
iface.tcp_wsf_state = <empty>
iface.tcp_wsf = 0
iface.tcp_timer_scale = 0
iface.tcp_timestamp = <empty>
iface.redirect = <empty>
iface.def_task_mgmt_timeout = 0
iface.header_digest = <empty>
iface.data_digest = <empty>
iface.immediate_data = <empty>
iface.initial_r2t = <empty>
iface.data_seq_inorder = <empty>
iface.data_pdu_inorder = <empty>
iface.erl = 0
iface.max_receive_data_len = 0
iface.first_burst_len = 0
iface.max_outstanding_r2t = 0
iface.max_burst_len = 0
iface.chap_auth = <empty>
iface.bidi_chap = <empty>
iface.strict_login_compliance = <empty>
iface.discovery_auth = <empty>
iface.discovery_logout = <empty>
node.discovery_address = 10.128.2.10
node.discovery_port = 3260
node.discovery_type = send_targets
node.session.initial_cmdsn = 0
node.session.initial_login_retry_max = 8
node.session.xmit_thread_priority = -20
node.session.cmds_max = 128
node.session.queue_depth = 32
node.session.nr_sessions = 1
node.session.auth.authmethod = None
node.session.auth.username = <empty>
node.session.auth.password = <empty>
node.session.auth.username_in = <empty>
node.session.auth.password_in = <empty>
node.session.auth.chap_algs = MD5
node.session.timeo.replacement_timeout = 120
node.session.err_timeo.abort_timeout = 15
node.session.err_timeo.lu_reset_timeout = 30
node.session.err_timeo.tgt_reset_timeout = 30
node.session.err_timeo.host_reset_timeout = 60
node.session.iscsi.FastAbort = Yes
node.session.iscsi.InitialR2T = No
node.session.iscsi.ImmediateData = Yes
node.session.iscsi.FirstBurstLength = 262144
node.session.iscsi.MaxBurstLength = 16776192
node.session.iscsi.DefaultTime2Retain = 0
node.session.iscsi.DefaultTime2Wait = 2
node.session.iscsi.MaxConnections = 1
node.session.iscsi.MaxOutstandingR2T = 1
node.session.iscsi.ERL = 0
node.session.scan = auto
node.session.reopen_max = 0
node.conn[0].address = 10.128.2.10
node.conn[0].port = 3260
node.conn[0].startup = manual
node.conn[0].tcp.window_size = 524288
node.conn[0].tcp.type_of_service = 0
node.conn[0].timeo.logout_timeout = 15
node.conn[0].timeo.login_timeout = 15
node.conn[0].timeo.auth_timeout = 45
node.conn[0].timeo.noop_out_interval = 5
node.conn[0].timeo.noop_out_timeout = 5
node.conn[0].iscsi.MaxXmitDataSegmentLength = 0
node.conn[0].iscsi.MaxRecvDataSegmentLength = 262144
node.conn[0].iscsi.HeaderDigest = None
node.conn[0].iscsi.DataDigest = None
node.conn[0].iscsi.IFMarker = No
node.conn[0].iscsi.OFMarker = No

# END RECORD
```

**Comment by netwarex** [comment]
@ArthurVardevanyan Do you have also the following error message in your dmesg?

```
[Sat Dec 10 18:52:01 2022] audit: type=1400 audit(1670698321.515:7214): avc:  denied  { dac_override } for  pid=231579 comm="iscsiadm" capability=1  scontext=system_u:system_r:iscsid_t:s0 tcontext=system_u:system_r:iscsid_t:s0 tclass=capability permissive=0
[Sat Dec 10 18:52:01 2022] audit: type=1300 audit(1670698321.515:7214): arch=c000003e syscall=83 success=no exit=-13 a0=55b9035185c0 a1=1f8 a2=ffffffffffffff00 a3=0 items=0 ppid=231163 pid=231579 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iscsiadm" exe="/usr/sbin/iscsiadm" subj=system_u:system_r:iscsid_t:s0 key=(null)
[Sat Dec 10 18:52:01 2022] audit: type=1327 audit(1670698321.515:7214): proctitle=697363736961646D002D6D00646973636F76657279002D740073656E6474617267657473002D700031302E3133312E312E31363
```

Happening after upgraded from `4.11.0-0.okd-2022-11-19-050030` to `4.11.0-0.okd-2022-12-02-145640`

Seems like some selinux permissions are not correct, however Longhorn didn't changed.

**Comment by ArthurVardevanyan** [comment]
@netwarex I do see the same thing in my dmesg, even on the latest CI Build, ```4.11.0-0.okd-2022-12-10-085957```.

The latest build of, ```4.12.0-0.okd-2022-12-10-034646```, seems to be working out of the box, so my current plan is to wait for Longhorn 1.4.0 then jump to  4.12 / 1.25.

**Comment by derekbit** [comment]
@ArthurVardevanyan @netwarex 
What's the version of your open-iscsi on the worker nodes? 
Can you help check `iscsiadm -V`? Thank you.

**Comment by netwarex** [comment]
@derekbit 
```
# iscsiadm -V
iscsiadm version 6.2.1.4
```

Also I was able to add the `dac_override` capability using a local CIL via MachineConfig, it seems to make this work now.

**Comment by derekbit** [comment]
@netwarex 

Thank you for the information.
Yes, adding `dac_override` is a workaround for this issue. (cc @ArthurVardevanyan)

`iscsi-initiator-utils 6.2.1.4` is based on `open-iscsi 2.1.4`. This issue appears to be caused by the permissions of the directories. It has been fixed in https://github.com/open-iscsi/open-iscsi/pull/244 and is included in `open-iscsi 2.1.4`.

Is your system is created by an older open-iscsi version prior to 2.1.4? If yes, you can check of the directories in the PR and correct their permissions to see if it can be fixed. It looks there is nothing that Longhorn can do to help with this issue from a Longhorn perspective.

What Longhorn can imporve is to stop the attach-detach loop and provide the appropriabe error message (ref: https://github.com/longhorn/longhorn/issues/5001).
Addtionally, we need a domain knowledge for the workaround.

cc @innobead

**Comment by netwarex** [comment]
@derekbit 
> `iscsi-initiator-utils 6.2.1.4` is based on `open-iscsi 2.1.4`. This issue appears to be caused by the permissions of the directories. It has been fixed in [open-iscsi/open-iscsi#244](https://github.com/open-iscsi/open-iscsi/pull/244) and is included in `open-iscsi 2.1.4`.

Yes, I ran into the same Github issue first, I didn't very yet the version, however this would be a regression as the 2 weeks before release with older FCOS does not have this issue, it worked just fine.

This could be the case, that they had the `dac_override` in the policies, but then they applied the fix and removed the policy... I currently don't have the time to reinstall an older version but this would make sense and would answer the issue.

However I'm not sure which folder is this to check, because I may just able to fix this manually with `chmod`, could you give a hint?

**Comment by derekbit** [comment]
> However I'm not sure which folder is this to check, because I may just able to fix this manually with chmod, could you give a hint?

Check the permissions of the folders such as LOCK_DIR and IFACE_CONFIG_DIR.
But, I think it is fine with the workaround adding `dac_override`.

**Comment by ArthurVardevanyan** [comment]
It is still interesting that is is able to mount using maintenance mode, but not regular mode.

@netwarex Do you have that yaml for the ```dac_override``` handy?

Some Additional Details below, unless I am checking the wrong process, it appears to have the ```dac_override``` already.

## 4.11.0-0.okd-2022-11-19-050030 (Working)
https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.11.0-0.okd-2022-11-19-050030
```bash
[core@worker-6 ~]$ iscsiadm -V
iscsiadm version 6.2.1.4

[core@worker-6 ~]$ systemctl status iscsid.service
[core@worker-6 ~]$ cat /proc/14827/status | grep Cap

CapInh:	0000000000000000
CapPrm:	000001ffffffffff
CapEff:	000001ffffffffff
CapBnd:	000001ffffffffff
CapAmb:	0000000000000000

[core@worker-6 ~]$ capsh --decode=000001ffffffffff
0x000001ffffffffff=cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read,cap_perfmon,cap_bpf,cap_checkpoint_restore

[core@worker-6 ~]$ ls -lsah /var/lib/iscsi/nodes/
total 4.0K
4.0K drwxr-xr-x. 7 root root 4.0K Dec 10 17:00 .
   0 drwxr-xr-x. 8 root root   90 Nov  2 20:41 ..
   0 drw-------. 3 root root   33 Dec 10 16:00 iqn.2019-10.io.longhorn:prometheus-k8s-db-prometheus-k8s-0

```

## 4.11.0-0.okd-2022-12-02-145640 (Not Working)
https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.11.0-0.okd-2022-12-02-145640
```bash
[core@worker-0 ~]$ iscsiadm -V
iscsiadm version 6.2.1.4

[core@worker-0 ~]$ systemctl status iscsid.service
[core@worker-0 ~]$ cat /proc/31881/status | grep Cap
CapInh:	0000000000000000
CapPrm:	000001ffffffffff
CapEff:	000001ffffffffff
CapBnd:	000001ffffffffff
CapAmb:	0000000000000000

[core@worker-0 ~]$ capsh --decode=000001ffffffffff
0x000001ffffffffff=cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read,cap_perfmon,cap_bpf,cap_checkpoint_restore

# Doesn't Mount, still shows up.
[core@worker-0 ~]$ ls -lsah /var/lib/iscsi/nodes/
total 0
0 drwxr-xr-x. 5 root root 116 Dec 11 14:52 .
0 drwxr-xr-x. 8 root root  90 Dec 11 14:24 ..
0 drw-------. 2 root root   6 Dec 11 14:52 iqn.2019-10.io.longhorn:stuck

```

## 4.12.0-0.okd-2022-12-10-034646 (Working)
https://amd64.origin.releases.ci.openshift.org/releasestream/4.12.0-0.okd/release/4.12.0-0.okd-2022-12-10-034646
```bash
[core@worker-2 ~]$  iscsiadm -V
iscsiadm version 6.2.1.4

[core@worker-2 ~]$ systemctl status iscsid.service
[core@worker-2 ~]$ cat /proc/22197/status | grep Cap
CapInh:	0000000000000000
CapPrm:	000001ffffffffff
CapEff:	000001ffffffffff
CapBnd:	000001ffffffffff
CapAmb:	0000000000000000

[core@worker-2 ~]$ capsh --decode=000001ffffffffff
0x000001ffffffffff=cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read,cap_perfmon,cap_bpf,cap_checkpoint_restore

[core@worker-2 ~]$ ls -lsah /var/lib/iscsi/nodes/
total 0
0 drwxr-xr-x. 5 root root 186 Dec 11 13:53 .
0 drwxr-xr-x. 8 root root  90 Dec 10 14:31 ..
0 drw-------. 3 root root  31 Dec 11 13:53 iqn.2019-10.io.longhorn:pvc-391ad221-81ea-4059-8b5e-2f060e527317
```

**Comment by netwarex** [comment]
@ArthurVardevanyan Sure, here it is:

https://gist.github.com/netwarex/5186be148560f99b5ac22759299e5407

Also can you please share the output of the policies on the same environments?

This is how you can extract as CIL:
```
cd /var/tmp
semodule -c -E iscsi
cat iscsi.cil
```

**Comment by ArthurVardevanyan** [comment]
> @ArthurVardevanyan Sure, here it is:
> 
> https://gist.github.com/netwarex/5186be148560f99b5ac22759299e5407
> 
> Also can you please share the output of the policies on the same environments?
> 
> This is how you can extract as CIL:
> 
> ```
> cd /var/tmp
> semodule -c -E iscsi
> cat iscsi.cil
> ```

Thank you for the commands:
[4.11.0-0.okd-2022-11-19-050030.log](https://github.com/longhorn/longhorn/files/10203490/4.11.0-0.okd-2022-11-19-050030.log)
[4.11.0-0.okd-2022-12-02-145640.log](https://github.com/longhorn/longhorn/files/10203489/4.11.0-0.okd-2022-12-02-145640.log)
[4.12.0-0.okd-2022-12-11-164611.log](https://github.com/longhorn/longhorn/files/10203512/4.12.0-0.okd-2022-12-11-164611.log)

I don't see a difference, however, my 4.11.0-0.okd-2022-12-02 Log is during a 4.11->4.12 Upgrade, but before the MCP Rollout.
I'll need to repave my sandbox later in the week, but I don't see ```dac_override``` with this command on either of the 3 versions.

**Comment by netwarex** [comment]
@ArthurVardevanyan Thanks for the logs. Then it's really a mistery how it went wrong... Maybe other policies somehow made this possible.

**Comment by ArthurVardevanyan** [comment]
OKD Refs: 
https://github.com/okd-project/okd/discussions/1430
https://github.com/okd-project/okd/issues/1438

**Comment by innobead** [comment]
@ChanYiLin could you help with this knowledge page as per https://github.com/longhorn/longhorn/issues/4988#issuecomment-1345476349? If any questions, feel free to check with @derekbit . Thanks.

**Comment by innobead** [comment]
Closing because the KB has been published. https://longhorn.io/kb/troubleshooting-volumes-stuck-in-attach-detach-loop-when-using-longhorn-on-okd/

**Comment by ArthurVardevanyan** [comment]
> Closing because the KB has been published. https://longhorn.io/kb/troubleshooting-volumes-stuck-in-attach-detach-loop-when-using-longhorn-on-okd/

Looks like the fixed version is no longer valid for point 1.

Nightlies were using a different base OS from Stable when 4.12 was released.


- https://github.com/longhorn/longhorn/pull/5004/commits/027ecca77d08b0eb6f3a4b1ac374d48a3758d16e
- https://github.com/okd-project/okd/issues/1438#issuecomment-1399317214

**Comment by innobead** [comment]
@ArthurVardevanyan thanks for the update.

**Comment by netwarex** [comment]
I have posted an article on this possible fix options, especially for the MachineConfig: https://ioflair.com/blog/fix-longhorn-volumes-stuck-in-attach-detach-loop-on-openshift-okd/

**Comment by ArthurVardevanyan** [comment]
Seems to be fixed in: 4.12.0-0.okd-2023-03-05-022504
